lib/absinthe/schema/notation/field_importer.ex

```diff
@@ -0,0 +1,93 @@
+defmodule Absinthe.Schema.Notation.FieldImporter do
+  def normalize_definitions(definitions) do
+    definitions_map = definitions |> build_def_map
+    errors = []
+    acc = []
+
+    import_context = %{
+      mapping: definitions_map,
+      parent: nil,
+      graph: :digraph.new([:acyclic]),
+    }
+
+    normalize(definitions, import_context, errors, acc)
+  end
+
+  def normalize([], _ctx, errors, acc) do
+    {:lists.reverse(acc), Enum.uniq(errors)}
+  end
+  def normalize([definition | rest], ctx, errors, acc) do
+    case do_normalize(definition, put_in(ctx.parent, definition)) do
+      {:ok, definition} ->
+        normalize(rest, ctx, errors, [definition | acc])
+      {:error, error} ->
+        normalize(rest, ctx, [error | errors], acc)
+    end
+  end
+
+  defp do_normalize(obj, ctx) do
+    _ = :digraph.add_vertex(ctx.graph, ctx.parent.identifier)
+    case obj.attrs[:field_imports] do
+      []->
+        {:ok, obj}
+      nil ->
+        {:ok, obj}
+      imports ->
+        with {:ok, fields} <- import_fields(imports, put_in(ctx.parent, obj), obj.attrs[:fields]) do
+          {:ok, %{obj | attrs: Keyword.update!(obj.attrs, :fields, fn _ -> fields end)}}
+        end
+    end
+  end
+
+  # Walk through the items we want to import fields from, get their fields,
+  # and walk to any objects THEY import to get their fields, etc.
+  defp import_fields(nil, _ctx, fields), do: {:ok, fields}
+  defp import_fields([], _ctx, fields), do: {:ok, fields}
+  defp import_fields([{obj_ref, _opts} | rest], ctx, existing_fields) do
```

Comment:
`opts` is here in the event that we want something like

```
import_fields :foo, only: [:foo,:bar,:baz]
# OR
import_fields :foo, except: [:blah]
```

Do you think that should be included now?